org.eclipse.jface.text/src/org/eclipse/jface/text/source/inlined/InlinedAnnotationSupport.java

```diff
@@ -545,8 +549,8 @@ private void removeInlinedAnnotations() {
 	 * @return the {@link AbstractInlinedAnnotation} from the given point and null otherwise.
 	 */
 	private AbstractInlinedAnnotation getInlinedAnnotationAtPoint(ISourceViewer viewer, int x, int y) {
-		if (fInlinedAnnotations != null) {
-			for (AbstractInlinedAnnotation ann : fInlinedAnnotations) {
+		if (fInlinedAnnotationByPositionMap != null) {
+			for (AbstractInlinedAnnotation ann : fInlinedAnnotationByPositionMap.values()) {
```

Comment:
Here it should interate on the map entries instead of values, and skip entries whose position isn't in the x/y range.

Reply:
I do not want to modify the old logic here, the goal was only to improve findExistingAnnotation() and adapt all the places where it was affected. 
Here I just replaced some lines so that it works with the new data structure the same as before. 
You can freely make modifications, commit and have someone else more knowledgeable here to review the code.
Also I think that it's not enough to verify that the position is in the x/y range because there are additional verifications  based on the actual text height/width annotation that's in the UI, see AbstractInlinedAnnotation.contains(). Also I would introduce code duplication if I would move the code from there to getInlinedAnnotationAtPoint().